app/routes/circulars/circulars.server.ts

```diff
@@ -263,3 +270,66 @@ export async function circularRedirect(query: string) {
     throw redirect(circularURL)
   }
 }
+
+async function* getAllRecords(): AsyncGenerator<Circular[], void, unknown> {
+  const data = await tables()
+  let exclusiveStartKey: AWS.DynamoDB.DocumentClient.Key | undefined = undefined
+
+  do {
+    const params: AWS.DynamoDB.DocumentClient.ScanInput = {
+      TableName: 'circulars',
+      ExclusiveStartKey: exclusiveStartKey,
+    }
+
+    const { Items, LastEvaluatedKey } = await data.circulars.scan(params)
+
+    if (Items) {
+      yield Items as Circular[]
+    }
+
+    exclusiveStartKey = LastEvaluatedKey
+  } while (exclusiveStartKey)
```

Comment:
Use a paginator rather than manually feeding the `LastEvaluatedKey` back to the API. There are several examples of scan paginators in our code that you can refer to.